object-cache.php

```diff
@@ -876,15 +914,52 @@ public function wp_action_admin_notices_warn_missing_apcu() {
 	 * @return null|WP_Object_Cache If cache is disabled, returns null.
 	 */
 	public function __construct() {
-		global $blog_id, $table_prefix, $wpdb;
+		global $blog_id, $table_prefix;
 
 		$this->multisite = is_multisite();
 		$this->blog_prefix = $this->multisite ? $blog_id . ':' : '';
 
-		$this->is_apcu_available = function_exists( 'apcu_sma_info' ) && apcu_sma_info();
+		// apcu_sma_info() triggers a warning when APCu is disabled
+		// @codingStandardsIgnoreStart
+		if ( function_exists( 'apcu_sma_info' ) && @apcu_sma_info() && class_exists( 'LCacheNullL1' ) ) {
+		// @codingStandardsIgnoreEnd
+			$l1 = new LCacheNullL1();
+			if ( php_sapi_name() !== 'cli' || 'on' === ini_get( 'apc.enable_cli' ) ) {
+				$l1 = new LCacheAPCuL1();
+			}
+			$port = null;
+			$socket = null;
+			$host = DB_HOST;
+			$port_or_socket = strstr( $host, ':' );
+			if ( ! empty( $port_or_socket ) ) {
+				$host = substr( $host, 0, strpos( $host, ':' ) );
+				$port_or_socket = substr( $port_or_socket, 1 );
+				if ( 0 !== strpos( $port_or_socket, '/' ) ) {
+					$port = intval( $port_or_socket );
+					$maybe_socket = strstr( $port_or_socket, ':' );
+					if ( ! empty( $maybe_socket ) ) {
+						$socket = substr( $maybe_socket, 1 );
+					}
+				} else {
+					$socket = $port_or_socket;
+				}
+			}
```

Comment:
I stole this out of `class wpdb`, but we might just want to use `$wpdb` instead.